CsharpExtrasTest/Extensions/ArrayExtensionTest.cs

```diff
@@ -377,18 +377,20 @@ public void Given_Array_When_To2DArray_Then_CorrectValueReturned(int[] array1D,
         }
 
         [Test]
-        [TestCaseSource("ProviderForDeepCopy")]
+        [TestCaseSource("ProviderForDeepCopy_With_Array")]
         public void Given_Array_When_DeepCopy_Then_CorrectValueReturned(int[] expectedArray)
         {
             //Act
             int[] actualArray = ArrayExtension.DeepCopy(expectedArray);
 
             //Assert
             Assert.AreEqual(expectedArray, actualArray);
+            expectedArray = new int[] { 4, 5, 6 };

```

Comment:
Not quite what I meant here. The purpose of the deep copy is to return an array that is **equal** to the input array, but not the **same** array. We need this because arrays are passed by reference, not by value. This means that if you pass an array into a function, that function can do something like `arr[1] = 10;` and modify the array the outer function is using.
To avoid this, a deep copy will create a new array object with the same contents as the original array. That way, if the copied array is modified, it does not change the original array.

My suggestion here would be to do something like:
```
Assert.AreEqual(expectedArray, actualArray);
actualArray[0] = actualArray[0] + 10; // Incrementing the value to make sure the value is different
Assert.AreNotEqual(expectedArray, actualArray, "A change to the copied array should not change the source array");
```